### ISSUE TITLE
Rethink/rewrite rule 953120 for re2 support 

### ISSUE THREAD

**Issue by fzipi** [coreruleset/coreruleset#2354]
### Proposed solution

The rule matches on the response body `<\?(?!xml)`, looking to detect the presence of the PHP open tag "`<?`" or "`<?php`" in output.

### Alternatives

Looking at [PHP tags](https://www.php.net/manual/en/language.basic-syntax.phptags.php) they can use `<\?\s+`, `<\?\=\s+` or `<\?php\s+` for the starting tag.

I think we should change it to match exactly for that, like: `<\?(?:\=|php)?\s+`.

**Comment by fzipi** [comment]
From `git blame`, I can see that `f173847a rules/RESPONSE-953-DATA-LEAKAGES-PHP.conf     (Federico G. Schwindt 2017-09-19 14:24:42 +0100  84) SecRule RESPONSE_BODY "@rx <\?(?!xml)" \` maybe @fgsch can add his thoughts here?

**Comment by fgsch** [comment]
I can't comment on the original intention of this rule as it's way before my time but the proposal looks sane (you don't need to escape `=` but that is cosmetic really).

**Comment by fzipi** [comment]
Excellent. Now, taking a look at the whole rule, to prevent false positives we are also chaining with ```SecRule RESPONSE_BODY "!@rx (?:\x1f\x8b\x08|\b(?:(?:i(?:nterplay|hdr|d3)|m(?:ovi|thd)|r(?:ar!|iff)|(?:ex|jf)if|f(?:lv|ws)|varg|cws)\b|gif)|B(?:%pdf|\.ra)\b|^wOF[F2])" \``` to not search for `<?` there. But, as I'm seeing it now, we don't need this if we use the space after to match. That won't happen on a binary file, IMHO. But we can keep that part if not sure.

@fgsch What do you think?

**Comment by fzipi** [comment]
Thanks. Updated PR and now is ready to be reviewed.

@lifeforms Can you take a look also to the linked PR?
